### PR TITLE
Watcher resource swagger specs. For more details on the Usage of the resource: https://docs.microsoft.com/en-us/azure/automation/automation-watchers-tutorial

### PR DESCRIPTION
**Pull request by mohanishpenta**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process.

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [x] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Comment by mohanishpenta** [comment]
Hey Laurent, Can you please take a look at this whenever you get a chance ? Thanks.

**Comment by mohanishpenta** [comment]
@ravbhatnagar - Can you please take a look at this whenever you get a chance ? thanks.

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json [hunk 1]
```diff
@@ -3830,6 +3830,160 @@
...
+      "description": {
+        "type": "string",
+        "description": "Gets or sets the watcher's description."
+      }
+    },
+    "description": "The properties of the create watcher operation."
+  },
+  "WatcherCreateOrUpdateParameters": {
+    "properties": {
+      "name": {
```
Comment (ravbhatnagar) [inline specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: name should not be updatable through the request body. read-only?

**Review by ravbhatnagar — COMMENTED** [review]
Just one comment. Looks like name is updatable in the request body. it should be marked as readonly. Other than that looks fine. Signing off from ARM side based on fixing this one comment

File: specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json (hunk 1 above)
Comment (mohanishpenta) [reply specification/automation/resource-manager/Microsoft.Automation/stable/2015-10-31/definitions.json]: Thanks for reviewing Gaurav. As discussed, using the same Watcher model for get/put and also using the same model of input and output of the put request. Marking fields in the watcher model as readonly as required. using a different model for input for patch.

**Comment by ravbhatnagar** [comment]
Looks good!

**Review by lmazuel — APPROVED** [review]
Merging, since PR content itself passes CI, even if global automation does not.

**Pull request merged**